go/azidext/azure_identity_credential_adapter.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidext
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// NewAzureIdentityCredentialAdapter is used to adapt an azcore.Credential to an autorest.Authorizer
+func NewAzureIdentityCredentialAdapter(credential azcore.Credential, options azcore.AuthenticationPolicyOptions) autorest.Authorizer {
+	policy := credential.AuthenticationPolicy(options)
+	return &policyAdapter{p: policy}
+}
+
+type policyAdapter struct {
+	p azcore.Policy
+}
+
+// WithAuthorization implements the autorest.Authorizer interface for type policyAdapter.
+func (ca *policyAdapter) WithAuthorization() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			r, err := p.Prepare(r)
+			if err != nil {
+				return r, err
+			}
+			_, err = ca.p.Do(r.Context(), &azcore.Request{Request: r})
```

Comment:
We recently refactored some public surface area in `azcore` which precludes the need to pass a context here.  Can you please update `azcore` to `v0.10.0` and fix this accordingly?